NewHaven/test/Part-3_4-Test.cpp

```diff
@@ -0,0 +1,55 @@
+//
+// Created by Muher on 2020-03-25.
+//
+#include <iostream>
+#include "Player.h"
+
+int main() {
```

Comment:
You shouldn't be creating a main, it wont compile/run without crashing anyways. Create a free function and assume everything already exists. You can create a `vector<player*>` globally and assume that it is initiated, but unknown length. Or just grab the players vector from `Setting::current` The work done here should really be just copy & paste into the driver for Part 3 without having to do any or light modifications.